python/datashaper/datashaper/graph.py

```diff
@@ -42,42 +43,77 @@ class ExecutionGraph:
     inputs: Dict[str, TableContainer] = {}
     __graph: Dict[str, ExecutionNode] = OrderedDict()
     __dependency_graph: Dict[str, set] = defaultdict(set)
+    __last_step_id: str = None
 
     def __init__(
-        self, workflow: Dict, input_path: str = "", schema_path: str = SCHEMA_FILE
+        self,
+        workflow: Dict,
+        input_path: Optional[str] = None,
+        input_tables: Optional[Dict[str, pd.DataFrame]] = None,
+        schema_path: Optional[str] = SCHEMA_FILE,
+        verbs: Optional[Dict[str, Callable]] = functions,
+        # TODO: the current schema definition does not work in Python
+        validate: bool = False,
     ):
         """Create an execution graph from the Dict provided in workflow.
 
         :param workflow: the Dict object that contains the workflow
         :type workflow: Dict
         :param input_path: Optional input path, if provided input
-                           tables will be loaded relative to that path, defaults to ""
+                           tables will be loaded relative to that path, defaults to None
         :type input_path: str, optional
+        :param schema_path: Optional Workflow schema path, if provided input
+                           tables will be loaded relative to that path, defaults to
+                           a known JSON schema path.
+        :type schema_path: str, optional
+        :param validate: Optional value, if true perform JSON-schema validation.
+                         Defaults to False.
+        :type validate: bool, optional
         """
-        with open(schema_path) as schema_file:
-            schema = json.load(schema_file)
-        validate(workflow, schema)
-
-        for input in workflow["input"]:
-            self.inputs[input] = TableContainer(
-                table=pd.read_csv(os.path.join(input_path, f"{input}.csv"))
-            )
-
+        # Perform JSON-schema validation
+        if validate:

```

Comment:
What happens here if validate is `True`, given the TODO comment above?

Reply:
It will blow up from the JsonSchema being invalid (there are some TS references in our schema - e.g. Record<string, something>, and Python doesn't appear to handle these well)

The validation is also problematic from the point of view that it's reaching outside of the library folder to find the default schema json file. In a publish version that would be a no-no.

Reply:
Validate seems to work in the tests - weird.